File: frontend/src/types/reservation.ts
export type ReservationResult = {
  reservationId: number;
  seatSlotId: number;
  seatId: number;
  userId: number;
  status: string;
  checkinCode: string;
  expiresAt: string;
  checkedInAt?: string | null;
  checkedOutAt?: string | null;
  lastWifiSeenAt?: string | null;
  seatLockQuota?: number | null;
  seatLockUsedCount?: number | null;
  lockedUntilAt?: string | null;
  seatNo?: string | null;
  seatLabel?: string | null;
  tableId?: number | null;
  tableNo?: string | null;
  areaId?: number | null;
  areaName?: string | null;
  floor?: string | null;
  slotDate?: string | null;
  startTime?: string | null;
  endTime?: string | null;
};

export type AdminSeatSlotReleaseResult = {
  seatSlotId: number;
  reservationId: number;
  releasedBy: number;
  reason: string;
  seatSlotStatus: string;
  reservation: ReservationResult;
};

export type AdminSeatSlotStatusResult = {
  seatSlotId: number;
  changedBy: number;
  reason: string;
  seatSlotStatus: string;
};

export type CheckinPayload = {
  checkinCode: string;
};

export type TableCheckinPayload = {
  tableQrToken: string;
  checkinCode: string;
};

export type WifiPresencePayload = object;

export type WifiPresenceResult = {
  reservationId: number;
  status: string;
  lastWifiSeenAt: string | null;
  offlineReleaseMinutes: number;
};

export type ReservationRule = {
  checkinGraceMinutes: number;
  checkinLeadMinutes: number;
  maxAdvanceDays: number;
  reservationOpenHour: number;
  dailyActiveReservationLimit: number;
  wifiOfflineReleaseMinutes: number;
  seatLockMinutes: number;
  updatedBy: number | null;
  updatedAt: string | null;
};

File: backend/src/main/java/com/lyston/smartseat/reservation/Reservation.java
package com.lyston.smartseat.reservation;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@TableName("reservations")
public class Reservation {

    @TableId
    private Long id;
    private Long userId;
    private Long seatId;
    private Long seatSlotId;
    private String status;
    private String checkinCode;
    private LocalDateTime reservedAt;
    private LocalDateTime checkedInAt;
    private LocalDateTime checkedOutAt;
    private LocalDateTime lastWifiSeenAt;
    private String lastWifiIp;
    private LocalDateTime expiresAt;
    @TableField(exist = false)
    private String seatNo;
    @TableField(exist = false)
    private String seatLabel;
    @TableField(exist = false)
    private Long tableId;
    @TableField(exist = false)
    private String tableNo;
    @TableField(exist = false)
    private Long areaId;
    @TableField(exist = false)
    private String areaName;
    @TableField(exist = false)
    private String floor;
    @TableField(exist = false)
    private LocalDate slotDate;
    @TableField(exist = false)
    private LocalTime startTime;
    @TableField(exist = false)
    private LocalTime endTime;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Long getSeatId() {
        return seatId;
    }

    public void setSeatId(Long seatId) {
        this.seatId = seatId;
    }

    public Long getSeatSlotId() {
        return seatSlotId;
    }

    public void setSeatSlotId(Long seatSlotId) {
        this.seatSlotId = seatSlotId;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getCheckinCode() {
        return checkinCode;
    }

    public void setCheckinCode(String checkinCode) {
        this.checkinCode = checkinCode;
    }

    public LocalDateTime getReservedAt() {
        return reservedAt;
    }

    public void setReservedAt(LocalDateTime reservedAt) {
        this.reservedAt = reservedAt;
    }

    public LocalDateTime getCheckedInAt() {
        return checkedInAt;
    }

    public void setCheckedInAt(LocalDateTime checkedInAt) {
        this.checkedInAt = checkedInAt;
    }

    public LocalDateTime getCheckedOutAt() {
        return checkedOutAt;
    }

    public void setCheckedOutAt(LocalDateTime checkedOutAt) {
        this.checkedOutAt = checkedOutAt;
    }

    public LocalDateTime getLastWifiSeenAt() {
        return lastWifiSeenAt;
    }

    public void setLastWifiSeenAt(LocalDateTime lastWifiSeenAt) {
        this.lastWifiSeenAt = lastWifiSeenAt;
    }

    public String getLastWifiIp() {
        return lastWifiIp;
    }

    public void setLastWifiIp(String lastWifiIp) {
        this.lastWifiIp = lastWifiIp;
    }

    public LocalDateTime getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(LocalDateTime expiresAt) {
        this.expiresAt = expiresAt;
    }

    public String getSeatNo() {
        return seatNo;
    }

    public void setSeatNo(String seatNo) {
        this.seatNo = seatNo;
    }

    public String getSeatLabel() {
        return seatLabel;
    }

    public void setSeatLabel(String seatLabel) {
        this.seatLabel = seatLabel;
    }

    public Long getTableId() {
        return tableId;
    }

    public void setTableId(Long tableId) {
        this.tableId = tableId;
    }

    public String getTableNo() {
        return tableNo;
    }

    public void setTableNo(String tableNo) {
        this.tableNo = tableNo;
    }

    public Long getAreaId() {
        return areaId;
    }

    public void setAreaId(Long areaId) {
        this.areaId = areaId;
    }

    public String getAreaName() {
        return areaName;
    }

    public void setAreaName(String areaName) {
        this.areaName = areaName;
    }

    public String getFloor() {
        return floor;
    }

    public void setFloor(String floor) {
        this.floor = floor;
    }

    public LocalDate getSlotDate() {
        return slotDate;
    }

    public void setSlotDate(LocalDate slotDate) {
        this.slotDate = slotDate;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalTime startTime) {
        this.startTime = startTime;
    }

    public LocalTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalTime endTime) {
        this.endTime = endTime;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
